pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -156,14 +158,15 @@ func New(options Options) Index {
 		XDSUpdater:                  options.XDSUpdater,
 		Debugger:                    options.Debugger,
 		Flags:                       options.Flags,
+		revision:                    options.Revision,
 		clientBuilder:               options.ClientBuilder,
 		stop:                        make(chan struct{}),
 		cs:                          multicluster.NewClustersStore(),
 		remoteClientConfigOverrides: options.RemoteClientConfigOverrides,
 	}
 
 	filter := kclient.Filter{
```

Comment:
I think we don't want this for Gateways. A user will set a rev on a gateway to make it programmed by istiod-A, but istiod-B needs to know about it.